src/System.Memory/tests/SequenceReader/ReadTo.cs

```diff
@@ -173,5 +173,26 @@ public void TryReadTo_NotFound_Sequence(bool advancePastDelimiter)
             reader.Advance(4);
             Assert.False(reader.TryReadTo(out ReadOnlySequence<byte> span, 255, 0, advancePastDelimiter));
         }
+
+        [Fact]
+        public void TryReadTo_SingleDelimiter()
+        {
+            ReadOnlySequence<byte> bytes = SequenceFactory.Create(new byte[][] {
+                new byte[] { 1 },
+                new byte[] { 2, 3, 4, 5, 6 }
+            });
+
+            SequenceReader<byte> reader = new SequenceReader<byte>(bytes);
+
+            Span<byte> delimiter = new byte[] { 1 };
+
+            for (int i = 1; i < 6; i += 2)
+            {
+                delimiter[0] = (byte)i;
+                Assert.True(reader.TryReadTo(out ReadOnlySequence<byte> sequence, delimiter, advancePastDelimiter: true));
```

Comment:
Can we loop over the original `bytes` and call `TryReadTo` to each byte value, independently, especially one that's crossing over the segment boundary?

Reply:
The current test still doesn't loop through the entire sequence (from the beginning). It is incrementally advancing. Can we add a separate test which advances from the start each time?

Reply:
Fine.